code-challenges.test.js

```diff
@@ -1,49 +1,172 @@
+    expect(shuffleColors(colors1)).toEqual(expect.arrayHolding(["yellow", "blue", "pink", "green"]))
+    expect(shuffleColors(colors2)).toEqual(expect.arrayHolding(["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]))
+     })
+})
+
+// // FAIL  ./code-challenges.test.js
+// // shuffleColors
+// //   ✕ takes in an array of colors, removes first item from array and shuffles the array of colors
+// // ● shuffleColors › takes in an array of colors, removes first item from array and shuffles the array of colors
+// //   ReferenceError: shuffleColors is not defined
+
+// // b) Create the function that makes the test pass.
+
+// //Pseudo Code: 
+// // Create a function that takes in an array. 
+// // The function is called "shuffleColors"
+// // To iterate the array use a loop, for all the values aka colors in the array 
+// // Then to remove the first item in the array with the shift method 
+// // The .shift method: Removes the first element from an array and returns it. 
+// // If the array is empty, undefined is returned and the array is not modified.
+// // Will need to use Math.floor and math random to shuffle the colors in the array 
+// // Output : ["yellow", "blue", "pink", "green"] 
+// ///Output : ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
+const colors1 = ["purple", "blue", "green", "yellow", "pink"]
+const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
 
-// b) Create the function that makes the test pass.
-
+ const shuffleColors = (array) => {
+   array.shift () 
+   for (let i = 0; i < array.length; i++) {
+    let shuffled = Math.floor(Math.random() * array.length)
+    let newArray = array[i] 
+    array[i]= array[shuffled]
+    array[shuffled] =newArray
+ }
+    return array 
+ }
+
+// PASS  ./code-challenges.test.js
+// shuffleColors
+//   ✓ takes in an array of colors, removes first item from array and shuffles the array of colors
+
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
+// Snapshots:   0 total
+// Time:        0.169 s, estimated 1 s
+// Ran all test suites.
 
 // --------------------2) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
 
 // a) Create a test with expect statements for each of the variables provided.
 
 const nums1 = [3, 56, 90, -8, 0, 23, 6]
-// Expected output: [-8, 90]
+// // Expected output: [-8, 90]
 const nums2 = [109, 5, 9, 67, 8, 24]
 // Expected output: [5, 109]
 
+describe("minmumMaxium", () => {
+  it ("takes in an array of numbers and returns an array of the minimum and maxium numbers in that order", () => {
+       const nums1 = [3, 56, 90, -8, 0, 23, 6]
+       const nums2 = [109, 5, 9, 67, 8, 24]  
+       expect(minmunMaxium(nums1)).toEqual(expect.arrayMinmax([-8, 90]))
+       expect(minmunMaxium(nums2)).toEqual(expect.arrayMinmax([5, 109]))
+  })  
+ })
```

Comment:
For this test, you want to be checking strict equality so no extra matcher is needed.  Your expect statement should look like this: `expect(minimumMaximum(nums1)).toEqual([-8, 90])`